src/index.ts

```diff
@@ -160,29 +164,57 @@ app.get('/browserAuthenticate.js', (req, res) => {
   res.sendFile(browserPath +'/dist/index.js');
 });
 
-app.get("/azureRedirect", async(req: any, res: any) => {
-  // The redirect will either contain a "code" or an "error"
+app.get("/startAzureRedirect",(req: express.Request,res: express.Response)=>{
+  const credential = new WebRedirectCredential( "27029f03-7c64-4ef6-88e4-14539e6c8d8c", 
+  "ed0e15fd-bb43-44e2-8d9c-b9c9758e0a46","http://localhost:8080/azureRedirect");
+  const state = (req.session as any).user;
+  const scope = "https://graph.microsoft.com/.default";
+  const authorizeUrl = credential.getAuthorizeUrl(scope,{state});
+  res.redirect(authorizeUrl);
+})
+
+app.get("/azureRedirect", async(req,res)=>{
   const authorizationCode = req.query["code"];
-  setLogLevel("verbose");
-  console.log(`authorization code = ${authorizationCode}`);
-  //for auth code cred, we need web for registering the redirect URIs
-  const authCred = new AuthorizationCodeCredential(
-    "27029f03-7c64-4ef6-88e4-14539e6c8d8c", 
-    "747a3a69-568f-4d40-9e9c-8f21472f246e",
-    authorizationCode,
-    "http://localhost:8080/azureRedirect")
+  if(!authorizationCode){
+    res.status(500).send('The authorization code is not found when attempting to login to azure');
+  }
+
+  const username = req.query["state"];
+  console.log(req.query);
+
+  const credential = new WebRedirectCredential( "27029f03-7c64-4ef6-88e4-14539e6c8d8c", 
+  "ed0e15fd-bb43-44e2-8d9c-b9c9758e0a46","http://localhost:8080/azureRedirect",{
+    //clientSecret: process.env.CLIENT_SECRET
+  });
+
+  const scope = "https://graph.microsoft.com/.default";
+
+  
   try{
-    const result = await authCred.getToken("https://graph.microsoft.com/.default");
+    console.log("First call to getToken");
+    const result = await credential.getToken(scope); //users might likely create an azure sdk client here instead of getToken directly
     console.log(result);
     res.send("Ok");
   }
   catch(e){
- console.error(e);
- res.send("not-ok");
+    console.log(e.message);
+    console.log(JSON.stringify(e));
+    if(e.name === "AuthenticationRequiredError"){
+      const authenticationRecord = await credential.authenticate(scope, authorizationCode as string);
+      (req.session as any).authenticationRecord = JSON.stringify(authenticationRecord);
```

Comment:
Why we pass in the `authorizationCode` to `authenticate` method instead of the `getToken` or constructor?
- Because we want to make the browser and node authentication experiences for JS developers similar.
- It gives more clarity about the purpose of `authorizationCode` and when it is used. Otherwise, it's usage is only hidden within the HTTP pipelines.
- `authenticate` also surfaces the usage of `authenticationRecord` which is very important for NodeJS developers because they have the necessity to tie the authenticated user to the current user session. And this enables them to save the user session in the database with the `authenticationRecord` type.